pkg/docker/docker.go

```diff
@@ -608,70 +606,81 @@ func (d *stiDocker) RunContainer(opts RunContainerOptions) error {
 
 	containerName := containerNameOrID(container)
 
-	// Container was created, so we defer its removal.
-	defer func() {
+	// Container was created, so we defer its removal, and also remove it if we get a SIGINT/SIGTERM/SIGQUIT/SIGHUP.
+	removalDone := make(chan struct{}, 1)
```

Comment:
With the current state of affairs we don't need this channel, right?

It is only sent to a few lines below, no other reference to it.